gc.c

```diff
@@ -5045,6 +5083,10 @@ static void invalidate_moved_page(rb_objspace_t *objspace, struct heap_page *pag
 static void
 read_barrier_handler(uintptr_t address)
 {
+    if (!is_pointer_to_heap_body(&rb_objspace, address)) {
```

Comment:
Why can't we use `is_pointer_to_heap` here?

Reply:
The segv handler returns the address exactly where the fault occurs - which is not necessarily aligned to an RVALUE. is_pointer_to_heap relies on the address being at the start of an RVALUE, otherwise it returns false.